### DISCUSSION TITLE
How to get the current context user name

### DISCUSSION THREAD

**Discussion by SP-Support** [pnp/powershell#578] [Q&A] [answered]
I run my PNP with my SharePoint Admin account as does the rest of my team. To make our scripts a bit more flexible can you explain how to get the current context user. If you use [Enivronment]::UserName it gives the computer user name not the user name the script is running as. Am I missing something?

**Comment by ToddKlindt** [comment, marked as answer]
I use this to see who is connected to M365, `(Get-PnPConnection | select -ExpandProperty pscredential).username`

**Reply by cwalexlai** [reply]
Would it work for MFA as well? e.g. Connect-PnPOnline -Interactive

**Reply by ToddKlindt** [reply]
I don't know, try it and let us know how it turns out. :)

**Reply by cwalexlai** [reply]
It's empty. I am unable to find a way to get current MFA logged in credentials.

**Reply by marcoscheel** [reply]
With a little bit of "csom" magic it is possible

```PowerShell
$ctx = Get-PnPContext
$ctx.Load($ctx.Web.CurrentUser)
$ctx.ExecuteQuery()
$ctx.Web.CurrentUser
$ctx.Web.CurrentUser.Email
```

**Reply by mpowney** [reply]
An alternative to the CSOM approach is to use the ```Get-PnPProperty``` command, with one-line:

```PowerShell
Get-PnPProperty -ClientObject (Get-PnPWeb) -Property CurrentUser
```

Behind the scenes, this works almost the same as the CSOM approach, but is not directly dependent on the CSOM ```ExecuteQuery``` method
